extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java

```diff
@@ -284,7 +285,7 @@ public void build(
         if (!applicationPaths.isEmpty()) {
             AnnotationInstance applicationPath = applicationPaths.iterator().next();
             rootPath = "/";
-            path = applicationPath.value().asString();
+            path = Encode.decode(applicationPath.value().asString());
```

Comment:
You expect the `applicationPath` to be encoded in the annotation? I wouldn't expect that. Could you clarify what you are trying to fix?

Reply:
Yes, the application path can be % encoded in the annotation:
https://github.com/jakartaee/rest/blob/e06f5fe15462550887967483b0114fd23f2c972f/jaxrs-api/src/main/java/jakarta/ws/rs/ApplicationPath.java#L45-L57

This change is to pass the TCK https://github.com/jakartaee/rest/blob/794354e9d40ea7d4a2eb47f822abd24c70c076f4/jaxrs-tck/src/main/java/ee/jakarta/tck/ws/rs/servlet3/rs/applicationpath/JAXRSClientIT
https://github.com/jakartaee/rest/blob/794354e9d40ea7d4a2eb47f822abd24c70c076f4/jaxrs-tck/src/main/java/ee/jakarta/tck/ws/rs/servlet3/rs/applicationpath/TSAppConfig.java#L25

Reply:
My, allowing both encoded and non-encoded values in the same place is really a good idea...